### DISCUSSION TITLE
How to react for a given entity descriptor instead of just components?

### DISCUSSION THREAD

**Discussion by boraty14** [sebas77/Svelto.ECS#125] [Q&A]
For the reactive interfaces for engines (IReactOnAddEx<T>, IReactOnSwapEx<T> etc) is it possible to react for a entity descriptor? In the system I want to react for the Descripted entity type (that contains multiple components), but interface accepts only Components as generics and not Descriptors.

@sebas77  recommended using group compounds but can you please give an example for it? Thanks

**Comment by sebas77** [comment]
Hi,

Unfortunately there isn't a svelto manual, so you must not miss the chance to learn new things reading all the articles and resources linked to Svelto as described in the wiki:

https://github.com/sebas77/Svelto.ECS/wiki

This article specifically will give a definition of GroupCompound: https://www.sebaslab.com/whats-new-in-svelto-ecs-3-0/

Once you have a Group compound defined like:

``` public class DOOFUSES : GroupTag<DOOFUSES> { }
        public class EATING : GroupTag<EATING> { }
        public class RED : GroupTag<RED> { }
        public class ENABLED : GroupTag<ENABLED> { }
        
            public class EatingRedDoofuses1  : GroupCompound<DOOFUSES, RED, EATING, ENABLED>{};
```

you can have IReactOnSwapEx callback like:

```
 public void MovedTo((uint start, uint end) rangeOfEntities, in EntityCollection<GameObjectEntityComponent> entities, ExclusiveGroupStruct fromGroup,
            ExclusiveGroupStruct toGroup)
        {
            if (DOOFUSES .Includes(fromGroup))
            {
//it's doofuses
            }
        }
```

you can see this logic being used in the MiniExample Survival, but very likely Doofuses has similar reasoning too.
